execution-engine/contracts/hdac-system/pop/src/contract_votes.rs

```diff
@@ -0,0 +1,226 @@
+                        write!(ret, "{:02x}", byte).expect("Writing to a string cannot fail");
+                    }
+                    ret
+                };
+
+                let user = to_hex_string(vote_key.user);
+                let dapp = to_hex_string_from_hash(vote_key.dapp);
+                let mut uref = String::new();
+                uref.write_fmt(format_args!("a_{}_{}_{}", user, dapp, balance))
+                    .expect("Writing to a string cannot fail");
+                uref
+            })
+            .collect();
+        // Remove and add urefs to update the contract's known urefs accordingly.
+        for (name, _) in runtime::list_named_keys() {
+            if name.starts_with("a_") && !new_urefs.remove(&name) {
+                runtime::remove_key(&name);
+            }
+        }
+        for name in new_urefs {
+            runtime::put_key(&name, Key::Hash([0; 32]));
+        }
+    }
+
+    #[allow(clippy::or_fun_call)]
+    pub fn read_stat() -> Result<VoteStat> {
+        let mut vote_stat = BTreeMap::new();
+        for (name, _) in runtime::list_named_keys() {
+            let mut split_name = name.split('_');
+            if Some("a") != split_name.next() {
+                continue;
+            }
+
+            let to_publickey = |hex_str: &str| -> Result<PublicKey> {
+                if hex_str.len() != 64 {
+                    return Err(Error::VoteKeyDeserializationFailed);
+                }
+                let mut key_bytes = [0u8; 32];
+                let _bytes_written = base16::decode_slice(hex_str, &mut key_bytes)
+                    .map_err(|_| Error::VoteKeyDeserializationFailed)?;
+                debug_assert!(_bytes_written == key_bytes.len());
+                Ok(PublicKey::from(key_bytes))
+            };
+
+            let to_hash = |hex_str: &str| -> Result<Key> {
+                if hex_str.len() != 64 {
+                    return Err(Error::VoteKeyDeserializationFailed);
+                }
+                let mut hash_bytes = [0u8; 32];
+                let _bytes_written = base16::decode_slice(hex_str, &mut hash_bytes)
+                    .map_err(|_| Error::VoteKeyDeserializationFailed)?;
+                debug_assert!(_bytes_written == hash_bytes.len());
+                Ok(Key::Hash(hash_bytes))
+            };
+
+            let hex_key = split_name
+                .next()
+                .ok_or(Error::VoteKeyDeserializationFailed)?;
+            let dapp_user = to_publickey(hex_key)?;
+
+            let hex_key = split_name
+                .next()
+                .ok_or(Error::VoteKeyDeserializationFailed)?;
+            let _dapp_owner = to_hash(hex_key)?;
+
+            let balance = split_name
+                .next()
+                .and_then(|b| U512::from_dec_str(b).ok())
+                .ok_or(Error::VotesDeserializationFailed)?;
+
+            let user_balance = vote_stat.entry(dapp_user).or_insert(U512::from(0));
```

Comment:
How about using `.or_insert_with(|| U512::from(0))` instead of suppressing clippy?

Reply:
https://github.com/hdac-io/CasperLabs/pull/14/commits/9464b5de93c7bca32387ba684abb64aee0814411